### DISCUSSION TITLE
Smart Start "Phantom Nodes" Not Removed

### DISCUSSION THREAD

**Discussion by The00Dustin** [zwave-js/zwave-js#6479] [Request Support / Investigate Issue] [answered]
### Checklist

- [X] I have read and followed the above instructions
- [X] I have checked the [troubleshooting section](https://zwave-js.github.io/node-zwave-js/#/troubleshooting/index) and my problem is **not** described there.

- [X] I have read the [changelog](https://github.com/zwave-js/node-zwave-js/blob/master/CHANGELOG.md) and my problem was **not** mentioned there or the fix **did not work**.

### Describe the issue

What is happening?
At approximately 22:48 UTC, I removed battery tabs from a Smart Start device and reassembled it, I noticed two new unidentified nodes in Z-Wave JS UI (45 & 46).  After waiting a bit expecting the second one to work, I triggered inclusion on the device (and not in Z-Wave JS UI) at approximately 22:50 UTC.  This lead to a new node that successfully interviewed.

What did you expect to happen instead?
I expected the first node ID to be re-used, but now I am under the impression that each node ID that wasn't successful should have been removed automatically.  I did use remove dead node to remove these, so it is possible that they could have been removed automatically if I had waited.

Anything else we should know?
Based on past experience with a mains-powered device after some cause of this behavior was resolved, I suspected that the key to triggering my experience was to remove power early in the interview of the Smart Start device.  As such, I suspect that the battery connection was not stable in this case until after I reassembled the device.  After all of this, I realized I needed to include with no security for association to another device, so the final node ID after disabling Smart Start is actually 48.

### Software versions

Driver (node-zwave-js): 12.2.1

Z-Wave JS UI: 9.2.3

Home Assistant Z-Wave Integration: 1.33.0

Home Assistant Z-Wave JS Addon: UI 2.2.3

### Device information

This issue doesn't seem to be device-specific.  Node IDs were mentioned in issue description.

### Checklist

- [X] I [made sure](https://zwave-js.github.io/node-zwave-js/#/troubleshooting/index?id=rule-1-use-the-driver-log-luke) to provide a **driver log** on level **debug**.

- [X] The log includes a re-interview of the problematic device _(if applicable)_.

- [X] The log includes the problematic interaction with the device _(if applicable)_.

- [X] I provided the node ID of the problematic device _(if applicable)_.

### Upload Logfile

[zwave-js-ui-store.zip](https://github.com/zwave-js/node-zwave-js/files/13214693/zwave-js-ui-store.zip)

**Comment by AlCalzone** [comment]
Thanks, I'll look into this. Something is preventing the existing code to remove failed SmartStart nodes from being executed.

**Comment by AlCalzone** [comment, marked as answer]
Fixed by https://github.com/zwave-js/node-zwave-js/pull/6483
